### DISCUSSION TITLE
Option to close zed instance if last window is closed

### DISCUSSION THREAD

**Discussion by DanielMSchmidt** [zed-industries/zed#8012] [Feature Requests]
Is there an option to close the zed instance if I press cmd+w again after closing the last window? I work with a lot of tmp dirs so having to switch to the mouse to close these windows out is a bit annoying. If it's not possible, any pointers on how one might implement this within zed?

**Comment by SomeoneToIgnore** [comment]
It should be possible and here's an issue for that even: https://github.com/zed-industries/zed/issues/5322
Since you're fine with pressing an extra keybindings, current workaround could be to use `cmd-shift-w` to close the window, before the feature is implemented.

As for general pointers, I would look at `workspace::CloseWindow` for the bits that close the window, and apply those on top of `pane::CloseActiveItem` logic that is responsible for closing tabs.
To be on the safe side, we'd better add another settings knob to enable/disable this behavior.

**Comment by JosephTLyons** [comment]
Hey @DanielMSchmidt this is now available within Zed - see:

- https://github.com/zed-industries/zed/pull/11989
